pkg/controller/azure/publicipaddress/actuator.go

```diff
@@ -279,7 +300,14 @@ func getFailedOperations(pubip *azurev1alpha1.PublicIPAddress) []azurev1alpha1.F
 }
 
 func shouldNotClean(pubip *azurev1alpha1.PublicIPAddress) bool {
-	return pubip.Annotations[DoNotCleanAnnotation] == strconv.FormatBool(true)
+	return pubip.Annotations[controllerazure.DoNotCleanAnnotation] == strconv.FormatBool(true)
+}
+
+func getServiceName(pubip *azurev1alpha1.PublicIPAddress) string {
```

Comment:
Just a minor thing. Not better to return here a `*string`? Then we could later check for `serviceName != nil` instead of `serviceName == ""`.

Reply:
@dkistner I can change it if you insist, but from my perspective it doesn't make much difference. `getServiceName` will never return an empty string if the pubip is associated with a service (an empty string is not a valid service name), so an empty string rather than a nil pointer should be perfectly ok here.

Reply:
As said just a nit. Up to you :)